vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -107,7 +108,9 @@ def _forward(
             query_pass = query[..., self.rotary_dim:]
             key_pass = key[..., self.rotary_dim:]
 
-        cos_sin = self.cos_sin_cache[positions]
+        self.cos_sin_cache = self.cos_sin_cache.to(positions.get_device())
```

Comment:
should use `positions.device` rather than `positions.get_device()`. 
https://pytorch.org/docs/stable/generated/torch.Tensor.get_device.html